app/models/ability.rb

```diff
@@ -76,9 +76,11 @@ def signed_in(user)
     can :manage, Event do |event|
       event.users.include?(user)
     end
-    # can create an event until the last day of a conference
-    can :create, Event, program_id: Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
 
+    # can create an event if program has call for papers defined
+    can [:new, :create], Event do |event|
+      event.program.cfp_open? && event.new_record?
```

Comment:
`can :manage, Event` covers all cases. `can [:new, :create], Event` [overwrites](https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence) this with rules not dependent on users.